Modules/Microsoft365DSC/DSCResources/MSFT_TeamsGroupPolicyAssignment/MSFT_TeamsGroupPolicyAssignment.psm1

```diff
@@ -0,0 +1,433 @@
+    }
+
+    #get groupid
+    if ($GroupId.Length -eq 0)
+    {
+        $Group = Find-CsGroup -SearchQuery $GroupDisplayName
+        if ($group.Length -gt 1)
+        {
+            Write-Verbose -Message "Found $($group.Length) groups with the name $GroupDisplayName"
+            $Group = $Group | Where-Object { $_.DisplayName -eq $GroupDisplayName }
+        }
+        if ($null -eq $Group)
+        {
+            Write-Verbose -Message "Group not found for $GroupDisplayName"
+            return
+        }
+        $GroupId = $Group.Id
+    }
+    Write-Verbose -Message "Retrieve GroupId for: $($GroupDisplayName)"
+    try
+    {
+        if ($Ensure -eq 'Present' -and $CurrentValues.Ensure -eq 'Absent')
+        {
+            Write-Verbose -Message "Adding GroupPolicyAssignment for $GroupDisplayName"
+            New-CsGroupPolicyAssignment -GroupId $GroupId -PolicyType $PolicyType -PolicyName $PolicyName -Rank $Priority
+        }
+        elseif ($Ensure -eq 'Present' -and $CurrentValues.Ensure -eq 'Present')
+        {
+            #Set-CsGroupPolicyAssignment not implemented jet / use remove-add as described in docs
+            Write-Verbose -Message "Remove GroupPolicyAssignment for $GroupDisplayName"
+            Remove-CsGroupPolicyAssignment -GroupId $CurrentValues.GroupId -PolicyType $CurrentValues.PolicyType
+            Write-Verbose -Message "Adding GroupPolicyAssignment for $GroupDisplayName"
+            New-CsGroupPolicyAssignment -GroupId $GroupId -PolicyType $PolicyType -PolicyName $PolicyName -Rank $Priority
+        }
+        elseif ($Ensure -eq 'Absent' -and $CurrentValues.Ensure -eq 'Present')
+        {
+            Write-Verbose -Message "Remove GroupPolicyAssignment for $GroupDisplayName"
+            Remove-CsGroupPolicyAssignment -GroupId $CurrentValues.GroupId -PolicyType $CurrentValues.PolicyType
+        }
+    }
+    catch
+    {
+        New-M365DSCLogEntry -Message "Error while setting GroupPolicyAssignment for $GroupDisplayName" `
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+    }
+}
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $GroupDisplayName,
+
+        [Parameter()]
+        [System.String]
+        $GroupId,
+
+        [Parameter()]
+        [ValidateSet('CallingLineIdentity', 'TeamsAppSetupPolicy', 'TeamsAudioConferencingPolicy', 'TeamsCallingPolicy', 'TeamsCallParkPolicy', 'TeamsChannelsPolicy', 'TeamsComplianceRecordingPolicy', 'TenantDialPlan', 'TeamsMeetingBroadcastPolicy', 'TeamsMeetingPolicy', 'TeamsMessagingPolicy', 'TeamsShiftsPolicy', 'TeamsUpdateManagementPolicy', 'TeamsVerticalPackagePolicy')]
+        [System.String]
+        $PolicyType,
+
+        [Parameter()]
+        [System.String]
+        $PolicyName,
+
+        [Parameter()]
```

Comment:
Now that Priority is a Key, we need to set it to Mandatory otherwise compilation will fail